### DISCUSSION TITLE
Directus 9, how can i set permissions on Files?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#7952] [Archive]
first of all i am so far totally happy withe the new directus 9, great work :)

I have an App with private messages , the messages are stored in a directus collections, each item has multiple Attachments (files). How can i restrict access to these files for other users, that are not in the Conversation?

(There are a conversations Collection wich have a one to many field to the message collection, wich have a multiple files field)
Thanks

**Comment by rijkvanzanten** [comment]
Thanks! ❤️ 

You _might_ be able to set up a many-to-many from files<->conversations, and then relationally follow the conversation to the user in the read permissions for directus_files. 

Assuming the following collection/field names:

```
directus_files.conversations (m2m) <-> files_conversations.file_id                        
                                       files_conversations.conversation_id <-> conversations.id
                                                                               conversations.users (m2m) <-> conversations_users.conversation_id
                                                                                                             conversations_users.user_id           <-> users.id
```

you might be able to follow that whole path back to the user with the following permissions for "read" on "directus_files": 

```json
{
   "conversations": {
      "conversation_id": {
         "users": {
            "user_id": {
               "_eq": "$CURRENT_USER"
            }
         }
      }
   }
}
```

(Saying "might" as this should theoretically work, though I've never gone this "deep" following a double-m2m relationship, so your mileage might vary 😄)

Alternatively, this open feature request for allow for permissions when you follow fields _relationally_ https://github.com/directus/directus/discussions/3144 could provide a way too 🤔

**Reply by ghost** [reply]
I ll try it :) thanks
